Basic flux structure

kusera opened the pull request.

Commit:
Add 'flux' to dependencies

Commit:
Add basic dispatcher

Commit:
Remove vanilla HTML prototype

Commit:
Add skeleton for Room flux parts

Commit:
Add object-assign polyfill dependency

Commit:
Add room creation functionality to RoomBox

chikamichi commented:
It's nice you opened a PR right away, made the review process quite easier on my side ;)

Commit:
Add ignore for local firebase json

Commit:
Add Firebase dependency

Commit:
Add skeleton for ChatAPI

Commit:
Change 'close' to 'delete'

Commit:
Add server action handler

- Changed VIEW-ACTION string to VIEW_ACTION

Commit:
Add API to handle firebase server interaction

Commit:
Add constant for room fetches

Commit:
Add action to dispatch room fetches

Commit:
Extract room conversion function from rendering

Commit:
Remove extraneous data attributes

Commit:
Add room fetch from server at app start

Commit:
Register response to room fetch

Commit:
Add constant for room submission

Commit:
Remove outdated properties for room creation

Commit:
Remove RoomActions require

Commit:
Cleaned up code organization

- Composite paths into single object
- Made error message slightly more descriptive

Commit:
Add skeleton for creating rooms on the firebase

- Made the language in RoomActions.js more consistent

Commit:
Change name of convertMessage()

Commit:
Moved RoomNode creation via object to RoomeNode

Commit:
Remove extra whitespace

Commit:
Listen to firebase for changes during first fetch

- Bug: ChatAPI fires off a ROOM_FETCHED dispatch before NEW_ROOM_SUBMIT
  finishes, throwing an error.
- Bug: RoomStore's setRoom() should be changed to replace its current list
  rather than appending to it (creates duplicate entries on subsequent
  fetches)

Commit:
Unify dispatch handlers and payload names

Commit:
Fix RoomStore not replacing its _rooms

Commit:
Remove outdated functions and dispatch registers

Commit:
Change 'submitRoom' to 'createRoom' everywhere

Commit:
Change _room clear to superior hack

Commit:
Listen to servers only once for getRoom

- Change 'err' to 'error' for future consistency

Commit:
Prevent room rewrites if it already exists

Commit:
Solve conflicting dispatch issue (for now)

- Split current and future server actions to a different Action Creator
- ChatAPI now "intercepts" certain actions from RoomActions, rather than
  listening for them by registering with Dispatcher
- Add placeholder function for room deletion pending component support

Commit:
Merge pull request #5 from kusera/firebase-api

Firebase API
